### PR TITLE
Adding Karma Unit Tests resolves #12 and #20 

### LINKED ISSUE #12: add unit tests
based off the work done in [**component-simple-slider**]
(https://github.com/ProjectEvergreen/component-simple-slider) to add unit tests, unit tests should be added to this project now.  

> ✨ _**Bonus points**:_ Submit back to the wiki docs if any interesting testing patterns / techniques are introduced.

related: https://github.com/ProjectEvergreen/project-evergreen/issues/1

### PR DESCRIPTION
**Pull request by hutchgrant**
## Related Issue
Resolves #12 and #20 

## Summary of Changes
- Updated to webpack-cli@3.1.1
- Added Karma configuration 
- Added tests for header and footer components

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    singleRun: shouldSingleRun,
+    captureTimeout: 210000,
+    browserDisconnectTolerance: 3,
+    browserDisconnectTimeout: 210000,
+    browserNoActivityTimeout: 210000,
+    concurrency: Infinity,
+    junitReporter: {
+      outputDir: './reports/test-results/',
+      outputFile: 'test-results.xml',
+      suite: 'my-app',
```
Comment (thescientist13) [inline karma.conf.js]: we could change this to _todo-app_.

File: src/components/todo-list/todo-list.js [hunk 2]
```diff
@@ -6,37 +6,44 @@ import '../todo-list-item/todo-list-item';
...
-    return { 
+  static get properties() {
+    return {
       todos: Array
     };
   }
 
-  addTodo() {
+  addTodo(e) {
+    e.preventDefault();
```
Comment (thescientist13) [inline src/components/todo-list/todo-list.js]: this looks like it includes changes from #22 ?  We should try and refrain from mixing changesets between branches as much as possible, e.g.  1 issue : 1 PR.

**Review by thescientist13 — COMMENTED** [review]
This is looking great @hutchgrant !

Just left a couple minor thoughts.  Also, we should refrain from introducing extensive formatting changes as part of PRs.  Until we automate it, it makes sense to keep what's there, which will help cut down on any extra diff noise when reviewing this PR.

File: src/components/todo-list/todo-list.js (hunk 2 above)
Comment (hutchgrant) [reply src/components/todo-list/todo-list.js]: You're right, I didn't even notice that.

File: package.json [hunk 3]
```diff
@@ -11,7 +11,8 @@
     "develop": "webpack-dev-server --config webpack.config.develop.js --open",
     "lint": "eslint *.js",
     "gh-pages": "rimraf ./docs && yarn build && cp -rv ./build/ ./docs",
-    "serve": "yarn build && cd build && http-server -o"
+    "serve": "yarn build && cd build && http-server -o",
+    "test": "karma start"
```
Comment (thescientist13) [inline package.json]: oh!  let's add `rimraf ./reports` here before `karma start` to clean out the _./reports_ directory for each run and then this is good to go!

**Review by thescientist13 — APPROVED** [review]
Thanks @hutchgrant , looks good!

**Pull request merged**